torch_xla/experimental/pjrt.py

```diff
@@ -311,3 +340,61 @@ def rendezvous(tag: str, payload: bytes,
   xm.mark_step()
 
   return [bytes(p.cpu().tolist()) for p in payloads]
+
+
+def init_pjrt_process_group(init_method: Optional[str] = None, **kwargs):
+  if not init_method and process_count() == 1:
+    init_method = f'file://{tempfile.mktemp()}'
+
+  dist.init_process_group(
+      'xla',
+      init_method,
+      rank=process_index(),
+      world_size=process_count(),
+      **kwargs)
+
+
+class DistributedDataParallel(nn.Module):
+  """Emulate DistributedDataParallel on TPUs.
+
+  Very experimental! There may still be correctness and performance issues to
+  work out. This class may be removed at any time.
+
+  torch.nn.parallel.DistributedDataParallel has additional overhead for gradient
+  bucketing that does not benefit TPUs. This implemenation may give better
+  performance on TPUs.
+
+  Compatible with multithreaded workloads required for multi-client execution on
+  TPU v2/v3.
+
+  Does not support model parallelism.
+  """
+
+  @staticmethod
+  def _reduce_grad(grad: torch.Tensor) -> torch.Tensor:
+    """Average gradients across replicas."""
+    return xm.all_reduce(xm.REDUCE_SUM, grad, scale=1. / global_device_count())
+
+  def __init__(self,
+               module: nn.Module,
+               *,
+               broadcast_buffers: bool = True,
```

Comment:
I guess we don't need to implement this logic?

Reply:
The `broadcast_buffers` is implemented in `forward`

Reply:
I guess my confusion is like if the previous optimizer_step approach doesn't demand this, what's the benefits of providing such option here? And then it doesn't seem like you test this option.